File: stack.py



class Stack:
    def __init__(self) -> None:
        self.head={}
    def push(self, value):
        self.head={value: value, next: self.head}

s = Stack()
s.push(6)
s.push(4)
s.push(5)
print('S:', s.head)

File: stack.js
class Stack {
	push(value) {
		this.head = { value, next: this.head };
	}
	pop() {
		let value;
		return ([value, this.head] = [this.head.value, this.head.next]), value;
	}
}

const s = new Stack();

s.push(3);
s.push(5);
console.log(s);

console.log(s.pop());
console.log(s.pop());
// console.log(s.pop());
